8Knot/pages/index/login_utils.py

```diff
@@ -0,0 +1,216 @@
+"""
+Login utilities
+Provides modular components for login banner and navigation elements.
+
+Generic Builders (All-in-One Pattern):
+    create_nav_item: Build any type of navigation item
+
+Login-Specific Functions:
+    create_login_navbar: Main login navigation bar
+    create_login_disabled_banner: Banner shown when login is disabled
+    is_login_enabled: Check login status from environment
+"""
+import os
+from typing import Optional, Dict, Any, Union, List, TYPE_CHECKING
+from dash import html, dcc
+import dash_bootstrap_components as dbc
+
+# Use TYPE_CHECKING to avoid circular imports
+if TYPE_CHECKING:
+    from db_manager.augur_manager import AugurManager
+
+
+# ===== GENERIC COMPONENT BUILDERS (All-in-One Pattern) =====
+
+
+def create_nav_item(
+    item_id: str,
+    content: str,
+    disabled: bool = False,
+    href: Optional[str] = None,
+    external_link: bool = False,
+    target: Optional[str] = None,
+) -> dbc.NavItem:
+    """
+    Generic navigation item builder following Dash All-in-One Components pattern.
```

Comment:
if this is meant to be an all in one component, it should probably be implemented as a class and be placed in its own file alongside our existing `VisualizationAIO` component.